packages/flutter/lib/src/widgets/app_lifecycle_listener.dart

```diff
@@ -0,0 +1,248 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+import 'package:flutter/foundation.dart';
+
+import 'binding.dart';
+
+/// A callback type that is used by [AppLifecycleListener.onExitRequested] to
+/// ask the application if it wants to cancel application termination or not.
+typedef AppExitRequestCallback = Future<AppExitResponse> Function();
+
+/// A listener that can be used to listen to changes in the application
+/// lifecycle.
+///
+/// To listen for requests for the application to exit, and to decide whether or
+/// not the application should exit when requested, create an
+/// [AppLifecycleListener] and set the [onExitRequested] callback. On macOS, in
+/// order to receive application exit requests originating from the OS, the
+/// `NSPrincipalClass` in your `Info.plist` file must be set to
+/// `FlutterApplication`, or a subclass of `FlutterApplication`.
```

Comment:
Is this set automatically if I do `flutter create` today? Or does it always require manual intervention?

Reply:
Oh! This is actually false now.  Removed.